src/lib/hash/hash.h

```diff
@@ -20,11 +20,30 @@ namespace Botan {
 class BOTAN_DLL HashFunction : public Buffered_Computation
    {
    public:
+      typedef SCAN_Name Spec;
```

Comment:
Without grepping in the source code, I had no clue that this is supposed to do. What about `using Spec = Botan::SCAN_Name;`?